app/src/main/java/com/belatrixsf/allstars/adapters/EmployeeListAdapter.java

```diff
@@ -32,37 +38,38 @@ public EmployeeListAdapter(Context context, List<Employee> employeeList) {
 
     @Override
     public EmployeeViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-
         View layoutView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_contact, parent, false);
-        EmployeeViewHolder rcv = new EmployeeViewHolder(layoutView);
-        return rcv;
+        return new EmployeeViewHolder(layoutView);
     }
 
     @Override
     public void onBindViewHolder(EmployeeViewHolder holder, int position) {
-        Employee emp = employeeList.get(position);
-
-        String fullName = emp.getFullName();
-        String letter;
-        if (null != fullName && !fullName.isEmpty()){
-            letter = String.valueOf(fullName.charAt(0));
-        }else{
-            letter = "#";
-        }
+        Employee employee = employeeList.get(position);
+        String fullName = employee.getFullName();
+        String currentFirstLetter = ( fullName != null && !fullName.isEmpty() ) ? String.valueOf(fullName.charAt(0)).toUpperCase() : "#";
+        String showLetter;
 
-        holder.firstLetter.setText(letter);
+        if ( lastFirstLetter == null ) showLetter = currentFirstLetter;
+        else showLetter = ( lastFirstLetter.equalsIgnoreCase(currentFirstLetter) ) ? "" : currentFirstLetter;
 
-        //holder.photo.setImageResource();
+        holder.firstLetter.setText(showLetter);
+        holder.fullName.setText(employee.getFullName());
+        holder.level.setText(String.valueOf(employee.getLevel()));
 
-        holder.fullName.setText(emp.getFullName());
-        holder.level.setText(String.valueOf(employeeList.get(position).getLevel()));
+        lastFirstLetter = currentFirstLetter;
     }
 
     @Override
     public int getItemCount() {
         return this.employeeList.size();
     }
 
+    public void updateData(List<Employee> employees){
+        employeeList.clear();
```

Comment:
how this will be handled with paging? Remember when there is more than 20 contacts you need to call page 2 of ws.